Root/OptimizationDump.cxx

```diff
@@ -377,13 +377,17 @@ EL::StatusCode OptimizationDump :: execute ()
   }
 
   if(!m_inputMET.empty() && !m_inputJets.empty()){
-    m_effectiveMass = VD::Meff_inclusive(in_met, in_jets, in_muons, in_electrons)/1000.;
+    
+    ConstDataVector<xAOD::JetContainer> presel_jets = VD::subset_using_decor(in_jets, VD::decor_vd_pass_Presel, 1);
+    //std::cout << in_jets->size() << " " << presel_jets.size() << std::endl;
+    m_effectiveMass = VD::Meff_inclusive(in_met, presel_jets.asDataVector(), in_muons, in_electrons)/1000.;
```

Comment:
Wrt your second point: if we add a pre-selection on m_eff, it should always be the signal objects for the analysis. So, "signal_pt_dependent_iso" for leptons, and "pass_preSel" for jets.Basically, we should always be using these objects for the calculations of the kinematic variables.